### ISSUE TITLE
Migrate fastlane setup from the Fabric Mac App to the recommended installation method

### ISSUE THREAD

**Issue by KrauseFx** [fastlane/fastlane#11586]
This doc is here for developers who have used the Fabric Mac App to install and update _fastlane_. To continue getting fastlane updates, you'll need to migrate to our preferred installation method for _fastlane_ by following these steps:

- `rm -rf ~/.fastlane/bin` (this will clear your current fastlane installation from your home directory)
- `rm -rf /usr/local/lib/fastlane_lib` (this will clear your current fastlane installation from your system)
- `rm /usr/local/bin/fastlane` (you might have a symlink here that needs to be removed)
These three items can be safely deleted, the Fabric Mac App is responsible for their existence.

- Edit your `~/.bashrc` or `~/.bash_profile` and remove any reference to `fastlane`
- Navigate to your project directory and run `touch Gemfile`, and add the following content to the file

```ruby
source "https://rubygems.org"

gem "fastlane"
```

**If you are using system ruby (and not [rbenv](https://github.com/rbenv/rbenv)):**
- Run `sudo gem install bundler`
- Run `sudo bundle update` 

**if you are using [rbenv](https://github.com/rbenv/rbenv) or other ruby manager:**
- Run `gem install bundler` 
- Run `bundle update` 

**For both methods of installation:**
- Add and commit both the `Gemfile` and the `Gemfile.lock` to version control (e.g. git)

From now on, every time you run _fastlane_, prefix your command using `bundle exec`, e.g.
```
bundle exec fastlane beta
```

This way, you'll always use the correct version of _fastlane_, and it's consistent across build machines and different work stations. 

To update _fastlane_ in the future, just run `fastlane update_fastlane`, and you're all set 🎉

**_Note:_** If you're currently running the Fabric Mac App, you'll need to restart the app after following these directions.

**Comment by viking2917** [comment]
After setting up the Gemfile, I attempt to run bundle and get

> bundle update
> -bash: bundle: command not found

What is bundle and where do I get it? (MacOS High Sierra 10.13.2). As I've borked my Fabric now I expect, and quick help much appreciated!

**Comment by viking2917** [comment]
(update:) 

`sudo gem install bundler
`
did the trick for me. never needed it before but I something has changed....

**Comment by mknippen** [comment]
Yes, I think bundler instructions should be included above. This is not something that comes on the Mac by default, or that most iOS devs use.

**Comment by mark-anders** [comment]
I too ran into bundle not found error and doing the `sudo gem install bundler` worked for me.

However, after doing that and performing `bundle update`, I got the following:

    Your user account isn't allowed to install to the system RubyGems.
      You can cancel this installation and run:

          bundle install --path vendor/bundle

      to install the gems into ./vendor/bundle/, or you can enter your password
      and install the bundled gems to RubyGems using sudo.

What is the recommended way?  It would be good to include those in the instructions.

**Comment by joshuafarray** [comment]
Rather than `sudo gem install bundler`, if you have Homebrew you can use [rbenv](https://github.com/rbenv/rbenv) pretty easily.

Here's a brief step-by-step that worked for me:

* Install rbenv: `brew install rbenv`
* Add rbenv to your environment by the following to ~/.bash_profile:
  `export PATH="/Users/username/.gem/ruby/2.3.0/bin:$PATH"`
  `eval "$(rbenv init -)"`
* Restart terminal
* Install a local ruby version: `rbenv install 2.3.1`  (this takes a while to complete)
* Activate the local ruby version:  `rbenv local 2.3.1`

Then you can `gem install bundler` without sudo.

**Comment by apwelsh** [comment]
Please DO NOT recommend we add homebrew to make this work.  It should work with a standard Mac development system.  The fact that these instructions still were not updated to provide instruction to install bundler, or that there are issues installing bundler is a joke, right?  How is it the the Fabric solution was painless, and worked w/o any custom tweaks, and this is the preferred method?  The purpose of these things is to make things easier, not harder and more cluttered.  

I am going to address this with an open mind, but if it gives me problems, I won't bother sticking with fastlane.  I only used it because twitterKit required Fabric, and this was kind of cool.  But, if it doesn't work, I'll happily go back to the very functional, and easy to use TestFlight.

**Comment by taquitos** [comment]
@bgolson, can you try both of these steps again?
- `rm -rf ~/.fastlane/bin` (this will clear your current fastlane installation from your home directory)
- `rm -rf /usr/local/lib/fastlane_lib` (this will clear your current fastlane installation from your system)

**Comment by bgolson** [comment]
@taquitos  I get the following now:

bundle exec fastlane update_fastlane
[✔] 🚀 
 -----------------------------
 --- Step: update_fastlane ---
 -----------------------------
 Looking for updates for fastlane...
 It seems that your Gem directory is not writable by your current User.
 fastlane would need sudo rights to update itself, however, running 'sudo fastlane' is not recommended.
 If you still want to use this action, please read the documentation on a guide how to set this up:
 https://docs.fastlane.tools/actions/#update_fastlane

Should I run 'sudo bundle exec fastlane update_fastlane' after deleting those files?

**Comment by bgolson** [comment]
sudo bundle exec fastlane update_fastlane
 -----------------------------
 --- Step: update_fastlane ---
 -----------------------------
 Looking for updates for fastlane...
 Nothing to update ✅
 
 Please help us test early releases of fastlane by opting into nightly builds 
 Just replace your `update_fastlane` call with
 
 ▸ update_fastlane(nightly: true)
 
 Nightly builds are reviewed and tested just like the public releases 🚂

**Comment by taquitos** [comment]
@bgolson, that looks good. Were you expecting an update to come through?

**Comment by bgolson** [comment]
Just re-opened the Fabric Mac app and it no longer shows the Fastlane tab, so I think I'm good to go now! Thanks so much @taquitos !

**Comment by superandrew** [comment]
it is unbelievable that I had a working system and now a broken one and I could choose between adding a rbenv I don't need (adding a fixed path to my .zshrc profile which is awkward and prone to future problems) OR use sudo everytime I have to launch `fastlane beta`. Wow. And now I will have to loose hours JUST to get back to a working fastlane. Which as someone else said, is very cool, but it is supposed to make things easier, not more complicated :/

**Comment by taquitos** [comment]
@superandrew spending hours on this is totally not our idea of easier 😞 You could also use `homebrew` to install fastlane and that should just work out of the box. We like to suggest `rbenv` because it helps keep your environments contained a bit, and generally reduces the amount of time you have to spend debugging weird gem issues, and _usually_ requires very little time to setup. Is there something we can help out with?

**Comment by superandrew** [comment]
when using rbenv one must add initialization and exporting variables in its shell profile. Which could work if not, as far as I understand, I must export the specific version of rbenv used when installing fastlane. 

But the main question is: if there is a brew option, since brew setup in my opinion is the easiest generally, why isn't that the preferred setup? 

The fabric setup has been lovely for its semplicity, now of course I totally understand the need of having a single codebase, but since this is a work tool, I'd suggest, before forcing users out of a working system into another, to find the simplest, always working way, and after a period of testing, prompt the majority of users when there's a reasonable certainty that it will work without hassle.

In my case I was able to fix it (the xcode schemes weren't shared, no idea why they worked before). Also, I was able to run it without sudoing, and that's ok.

Apologies for my complaints but I love this tool and honestly I didn't appreciate the way this forced update/upgrade was handled (e.g. using a GitHub issue).

Will try the brew setup and then eventually post the steps. How can I be sure I removed the current system ruby version?

**Comment by sureshjoshi** [comment]
Thanks for the instructions guys! Just a note, I needed to also run this command, since I had something in my local/bin directory

    rm /usr/local/bin/fastlane

**Comment by taquitos** [comment]
@sureshjoshi thanks for the note, I updated the instructions to mention that a symlink might be there 👍

**Comment by exavi** [comment]
Running all of the above steps broke completely my fastlane install and now I've reverted to doing everything the manual way.

Using the system ruby version I got this error when running the command
```sudo bundle update```
error message:
```
Installing jwt 2.1.0
Gem::InstallError: jwt requires Ruby version >= 2.1.
An error occurred while installing jwt (2.1.0), and Bundler cannot
continue.
Make sure that `gem install jwt -v '2.1.0'` succeeds before bundling.

In Gemfile:
  fastlane was resolved to 2.92.1, which depends on
    google-api-client was resolved to 0.13.6, which depends on
      googleauth was resolved to 0.6.2, which depends on
        signet was resolved to 0.8.1, which depends on
          jwt

```
tried installing rbenv and running that mess separately, which resulted in rbenv not really working and trying to install stuff to the system provided ruby

```
gem install bundler
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.0.0 directory.
```

If I run it with sudo, it succeeds but since it's using the system provided paths it will eventually error with jwt too.

I tried installing with brew (which was the way I had it installed, waaay before the fabric mac app *forced* me to install it through itself, and I had to remove the install and reinstall, after a "successful" reinstall, I can't seem to be able to run fastlane yet, this is what I get:

```
[10:45:55]: Successfully installed 'cocoapods'
[10:45:55]: $ bundle exec pod install
[10:45:56]: ▸ bundler: failed to load command: pod (/Users/exavi/.fastlane/bin/bundle/bin/pod)
[10:45:56]: ▸ Gem::Exception: can't find executable pod
[10:45:56]: ▸ /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/rubygems_integration.rb:408:in `block in replace_bin_path'
[10:45:56]: ▸ /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/rubygems_integration.rb:434:in `block in replace_bin_path'
[10:45:56]: ▸ /Users/exavi/.fastlane/bin/bundle/bin/pod:22:in `<top (required)>'
[10:45:56]: Exit status of command 'bundle exec pod install' was 1 instead of 0.
bundler: failed to load command: pod (/Users/exavi/.fastlane/bin/bundle/bin/pod)
Gem::Exception: can't find executable pod
  /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/rubygems_integration.rb:408:in `block in replace_bin_path'
  /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/rubygems_integration.rb:434:in `block in replace_bin_path'
  /Users/exavi/.fastlane/bin/bundle/bin/pod:22:in `<top (required)>'

+------------------+--------------+
|          Lane Context           |
+------------------+--------------+
| DEFAULT_PLATFORM | ios          |
| PLATFORM_NAME    | ios          |
| LANE_NAME        | ios appstore |
+------------------+--------------+
[10:45:56]: bundler: failed to load command: pod (/Users/exavi/.fastlane/bin/bundle/bin/pod)
Gem::Exception: can't find executable pod
  /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/rubygems_integration.rb:408:in `block in replace_bin_path'
  /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/rubygems_integration.rb:434:in `block in replace_bin_path'
  /Users/exavi/.fastlane/bin/bundle/bin/pod:22:in `<top (required)>'


+------+---------------------+-------------+
|             fastlane summary             |
+------+---------------------+-------------+
| Step | Action              | Time (in s) |
+------+---------------------+-------------+
| 1    | Verifying fastlane  | 0           |
|      | version             |             |
| 2    | default_platform    | 0           |
| 💥   | cocoapods           | 11          |
+------+---------------------+-------------+

[10:45:56]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

/Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane_core/lib/fastlane_core/ui/interface.rb:145:in `shell_error!': [!] bundler: failed to load command: pod (/Users/exavi/.fastlane/bin/bundle/bin/pod) (FastlaneCore::Interface::FastlaneShellError)
Gem::Exception: can't find executable pod
  /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/rubygems_integration.rb:408:in `block in replace_bin_path'
  /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/bundler-1.14.6/lib/bundler/rubygems_integration.rb:434:in `block in replace_bin_path'
  /Users/exavi/.fastlane/bin/bundle/bin/pod:22:in `<top (required)>'

	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/actions/cocoapods.rb:45:in `call_error_callback'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/actions/cocoapods.rb:34:in `block in run'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/helper/sh_helper.rb:78:in `call'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/helper/sh_helper.rb:78:in `sh_control_output'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/helper/sh_helper.rb:12:in `sh'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/actions/cocoapods.rb:27:in `run'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:257:in `block (2 levels) in execute_action'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:236:in `block in execute_action'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:231:in `chdir'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:231:in `execute_action'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
	from Fastfile:21:in `block (2 levels) in parsing_binding'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:289:in `call'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:289:in `execute_flow_block'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:46:in `block in execute'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/lane_manager.rb:59:in `cruise_lane'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/fastlane/lib/fastlane/cli_tools_distributor.rb:107:in `take_off'
	from /Users/exavi/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.1/bin/fastlane:20:in `<top (required)>'
	from /Users/exavi/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/exavi/.fastlane/bin/bundle/bin/fastlane:22:in `<main>'

```

I'm a bit desperate, fastlane is so good that I now have come to rely heavily on it, is there any way that I can just install even if it's an older version of it? I would really like to go back to my older version, I don't mind that it is outdated, it just worked, I'll sort something out when it's absolutely necessary to upgrade, in the meantime I need whatever was there before this broken upgrade.

**Comment by exavi** [comment]
Small update, this is probably a horrible workaround but worked for me, after trying the three methods above.

I installed the latest version of ruby through brew, edited my $PATH to point to brew's version earlier in the binary searchpath, tried to install bundler again and it all worked.

I have just submitted a new version of an app through this newly installed fastlane and everything seems good, I'm currently testing it through Testflight, seems ok.
